publish-extensions.js

```diff
@@ -135,8 +136,19 @@ const flags = [
       }
 
       let timeout;
-      await new Promise((resolve, reject) => {
-        const p = cp.spawn(process.execPath, ['publish-extension.js', JSON.stringify(extension)], {
+      await new Promise(async (resolve, reject) => {
+        const repository = extension.download && extension.download.replace(/\/releases\/download\/.*$/, '');
+        if (msVersion && (!repository && !extension.repository)) {
+          reject(new Error("Repository not found, you should add it to extensions.json"));
+        }
+
+        const vsixDownloadLink = extension.download && await getReleases.findLatestVSIXRelease(repository || extension.repository, extension.version, msVersion);
+        if(repository && !vsixDownloadLink) {
```

Comment:
We should fail if repository is not provided before trying to check for release no?